assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/stable_diffusion/mlflow_wrapper.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""
+Contains MLFlow pyfunc wrapper for stable diffusion models.
+Has methods to load the model and predict.
+"""
+
+import base64
+from diffusers import StableDiffusionPipeline
+import pandas as pd
+from io import BytesIO
+import mlflow
+import torch
+
+from .constants import COLUMN_NAMES, DATATYPE_LITERALS, Tasks
+
+
+class StableDiffusionMLflowWrapper(mlflow.pyfunc.PythonModel):
+    """MLflow model wrapper for stable diffusion models."""
+
+    def __init__(
+            self,
+            task_type: str,
+            model_id: str,
+    ) -> None:
+        """Constructor for MLflow wrapper class
+
+        :param task_type: Task type used in training.
+        :type task_type: str
+        :param model_id: Hugging face model id of stable diffusion models.
+        :type model_id: str
+        """
+        super().__init__()
+        self._pipe = None
+        self._task_type = task_type
+        self._model_id = model_id
+
+    def load_context(self, context: mlflow.pyfunc.PythonModelContext) -> None:
+        """
+        Load a MLflow model with pyfunc.load_model().
+
+        :param context: MLflow context containing artifacts that the model can use for inference
+        :type context: mlflow.pyfunc.PythonModelContext
+        """
+        if self._task_type in [Tasks.TEXT_TO_IMAGE, Tasks.IMAGE_INPAINTING]:
+            try:
+                _map_location = "cuda" if torch.cuda.is_available() else "cpu"
+
+                self._pipe = StableDiffusionPipeline.from_pretrained(self._model_id)
+                self._pipe.to(_map_location)
```

Comment:
please verify if it's working with multi-gpu scenarios.

Reply:
Good point, I'll test multi-gpu in next iteration.